src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,415 @@
+﻿// Licensed to the.NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.I2c;
+using UnitsNet;
+
+namespace Iot.Device.UFire
+{
+    /// <summary>
+    /// μFire ISE (Ion Specific Electrode) Probe Interface controller
+    /// </summary>
+    public class UFireIse : IDisposable
+    {
+        /// <summary>
+        /// 0x3F is the default address of all sensors
+        /// </summary>
+        public static byte I2cAddress = 0x3F;
+
+        // A temperature measurement takes 750ms, see https://www.ufire.co/docs/uFire_ISE/#use
+        private const int IseTemperatureMeasureTime = 750;
+        // A measurement takes 750 ms, see https://www.ufire.co/docs/uFire_ISE/#use
+        private const int IseMvMeasureTime = 750;
+
+        // Implement short delays, required for the device to operate reliably, see https://www.ufire.co/docs/uFire_ISE/#use
+        private const int IseCommunicationDelay = 10;
+
+        private const bool IseDualPointConfigBit = false;
+        private const int IseTemperatureCompensationConfigBit = 1;
+
+        private I2cDevice _device;
+
+        private float _measurement = 0;
+        private Temperature _temperature = new Temperature();
+
+        private bool _temperatureCompensation = false;
+
+        /// <summary>
+        /// Use temperature compensation
+        /// </summary>
+        public bool TemperatureCompensation
+        {
+            get
+            {
+                return _temperatureCompensation;
+            }
```

Comment:
nit: this can be written as:

```suggestion
            get => _temperatureCompensation;
```